lib/path.js

```diff
@@ -115,8 +138,8 @@ win32.resolve = function() {
 
     var result = splitDeviceRe.exec(path),
         device = result[1] || '',
-        isUnc = device && device.charAt(1) !== ':',
-        isAbsolute = win32.isAbsolute(path),
+        isUnc = device && device[1] !== ':',
+        isAbsolute = isUnc || result[2],
```

Comment:
Unless `win32.isAbsolute` is on the critical path of a very common use case in terms of performance, I have a preference for using a helper function such as the one you described [in your comment above](https://github.com/joyent/node/pull/9289#discussion_r26546997). 

I want to avoid subtle regressions due to potential divergence between several versions of inlined `win32.isAbsolute`.